content/en/docs/concepts/extend-kubernetes/compute-storage-net/resource-drivers.md

```diff
@@ -0,0 +1,280 @@
+
+- a controller (one per cluster), manages hardware resources allocation for
+  {{< glossary_tooltip term_id="resource-claim" text="Resource Claims">}}
+- kubelet-plugin (one per node with resource), that:
+  - discovers the supported hardware
+  - announces the discovered hardware to the Resource Driver controller
+  - prepares the hardware allocated for Pod when Pod comes to the Kubelet
+  - unprepares the hardware allocated for a Pod when the Pod has reached final state or is being deleted.
+
+Communication between the controller and the kubelet-plugin can be realized in two ways:
+
+- through Custom Resource Definition objects provided by and designed for the resource driver
+- through a ResourceHandle which is a part of an `AllocationResult` provided by the controller in case of
+  successful allocation
+
+General recommendations:
+
+- resource driver name pattern: `<HW type>.resource.<companyname>.<companydomain>`, e.g.
+  gpu.resource.example.com
+
+<!-- body -->
+
+## Resource driver registration
+
+Controller and kubelet-plugin registration can be handled with the DRA helper code that implements
+common for all resource drivers functionality.
+
+### Controller registration
+
+Resource driver controller has to implement a [driver interface](https://pkg.go.dev/k8s.io/dynamic-resource-allocation@v0.28.1/controller#Driver),
+which then can be simply used with [controller helper instance](https://pkg.go.dev/k8s.io/dynamic-resource-allocation@v0.28.1/controller#New).
+
+See [example resource driver controller](https://github.com/kubernetes-sigs/dra-example-driver/blob/151b7c8da2e620c47a3591e1a937f8d0297b0c25/cmd/dra-example-controller/main.go#L208) for details
+
+
+### Kubelet-plugin registration
+
+Resource driver kubelet-plugin has to implement [NodeServer interface](https://pkg.go.dev/k8s.io/kubelet@v0.28.2/pkg/apis/dra/v1alpha3#NodeServer),
+which can be simply used with [kubelet-plugin helper instance](https://github.com/kubernetes-sigs/dra-example-driver/blob/151b7c8da2e620c47a3591e1a937f8d0297b0c25/cmd/dra-example-kubeletplugin/main.go#L160).
+
+## Controller implementation
+
+Controller's main responsibility is to allocate and deallocate resource for
+{{< glossary_tooltip term_id="resource-claim" text="Resource Claim">}}.
+
+There are two modes of allocation the Resource Claim can have:
+
+- `WaitForFirstConsumer` (default), AKA "delayed". In this mode the resource will only be requested
+  for Resource Claim when a Pod that needs it is being scheduled.
+- "Immediate": the resource has to be allocated to the Resource Claim as soon as possible, and
+  retained until the Resource Claim is deleted.
+
+If `allocationMode` is not set explicitly, the mode is `WaitForFirstConsumer`.
+
+### Delayed allocation
+
+Delayed allocation is done in two steps: first, the scheduler picks up a list of Node names and
+asks from resource driver controller which of them are unsuitable for Resource Claim through
+`UnsuitableNodes` call.
+
```

Comment:
It would be nice if this doc could provide more context on what will happens if there is no remain node after the filter plugin. (I'm talking about PostFilter)

Reply:
Done.